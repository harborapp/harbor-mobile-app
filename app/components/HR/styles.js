import { StyleSheet } from 'react-native'

import * as colors from '../../constants/colors'

export default StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    marginHorizontal: 8
  },
  line: {
    flex: 1,
    height: 1,
    backgroundColor: colors.LOGIN_OPTIONS_SEP
  },
  text: {
    marginLeft: 15,
    marginRight: 15
  }
})
