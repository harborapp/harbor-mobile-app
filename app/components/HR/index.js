import React from 'react'
import PropTypes from 'prop-types'
import { View } from 'react-native'

import styles from './styles'

/**
 * HR is used to render two horizontal lines with text in between.
 * It makes no assumptions about how the text is shown (for e.g
 * with a Formatted Intl message instead of Text) by consuming a
 * child as a function.
 *
 * @returns {jsx}
 *
 */
const HR = ({ children }) => (
  <View style={styles.container}>
    <View style={styles.line} />
    <View>{children(styles.text)}</View>
    <View style={styles.line} />
  </View>
)

HR.propTypes = {
  children: PropTypes.func.isRequired
}

export default HR
