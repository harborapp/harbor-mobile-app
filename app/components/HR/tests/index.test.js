/* eslint-env jest */
import React from 'react'
import { Text } from 'react-native'
import renderer from 'react-test-renderer'

import HR from '../index'

describe('HR', () => {
  it('should render', () => {
    const tree = renderer.create(
      <HR>
        {(style) => <Text style={style}>Hello</Text>}
      </HR>
    )

    expect(tree).toMatchSnapshot()
  })
})
