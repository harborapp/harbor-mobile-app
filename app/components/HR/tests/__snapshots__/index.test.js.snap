// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HR should render 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginHorizontal": 8,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "rgba(255, 255, 255, 0.3)",
        "flex": 1,
        "height": 1,
      }
    }
  />
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "flex": 1,
          "marginLeft": 15,
          "marginRight": 15,
          "marginTop": 5,
          "textAlign": "center",
        }
      }
    >
      Hello
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "rgba(255, 255, 255, 0.3)",
        "flex": 1,
        "height": 1,
      }
    }
  />
</View>
`;
